xbmc/music/tags/TagLibVFSStream.cpp

```diff
@@ -37,6 +39,17 @@ TagLibVFSStream::TagLibVFSStream(const std::string& strFileName, bool readOnly)
   {
     if (!m_file.Open(strFileName))
       m_bIsOpen = false;
+    else
+    { // Check if this stream is a Shoutcast, and skip scanning tags if it is.
+      // Shoutcast has its own metadata format that TagLib does not read
+      CFileCache* fileCacheItem = dynamic_cast<CFileCache*>(m_file.GetImplementation());
```

Comment:
see http://kodi.wiki/view/advancedsettings.xml#cache

maybe this could work:
try cast to filecache, if that works cast this into shoutcast, else cast original into shoutcast